cvmfs/util/posix.cc

```diff
@@ -1750,6 +1750,70 @@ struct ForkFailures {  // TODO(rmeusel): C++11 (type safe enum)
   }
 };
 
+/**
+ * Closes all file descriptors except the ones in preserve_fildes.
+ * To be used after fork but before exec.
+ */
+static bool CloseAllFildes(const std::set<int> &preserve_fildes)
+{
+  int max_fd = static_cast<int>(sysconf(_SC_OPEN_MAX));
+  if (max_fd < 0) {
+    return false;
+  }
+
+  auto close_all_until_maxfd = [&preserve_fildes](int max_fd){
```

Comment:
Cvmfs is C++03, this won't compile on all supported platforms.

Reply:
Ah ok, I'll redo it for that.